Fix broken dfn "agreement" and update language indexing

aphillips opened the pull request:
Match changes to specdev
Add a dfn for `agreement`/`serialization agreement` which somehow got lost and is making respec unhappy.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/string-meta/pull/78.html" title="Last updated on Jul 13, 2023, 3:11 PM UTC (7806817)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/78/3ed0424...aphillips:7806817.html" title="Last updated on Jul 13, 2023, 3:11 PM UTC (7806817)">Diff</a>

aphillips committed:
Fix broken dfn "agreement" and update language indexing
- Make the text about language indexing mirror the changes in specdev
- Add a `dfn` for term "agreement" (supporting also "serialization
  agreement") so that ReSpec is happy

aphillips committed:
Fix typo :-P

xfq commented:
Not sure why Netlify doesn't work anymore. I just reinstalled it. We can push a new commit and see if it works.

aphillips committed:
Additional edits per 2023-06-15 telecon
- Rewrote the section to better separate language negotiation from
  language indexing and to indicate some of the cons of language
  indexing.

aphillips committed:
Fix broken term 'agreement'

r12a commented on index.html:
```diff
@@ -1163,9 +1163,13 @@ <h4 id="language-detection-issues">Issues</h4>
+	
+	<p>In these cases, localization of a content item is done by having the <a>producer</a> return multiple language representations for the item and letting the <a>consumer</a> choose the value to display. Such an approach is helpful when the <a>producer</a> cannot negotiate the language (such as when the resulting file is cached for multiple users) and when the number of languages is relatively sm ...
+	
+	<p>One approach a specification might provide for returning mulitple language of a given field is called <dfn>language indexing</dfn>. In language indexing, a given field's value is an array of key-value pairs. The keys in the array are language tags. The values of each language tag are strings or, ideally, <a>Localizable</a> objects. Here's an example of what a language indexed field <kbd>title ...
```
mulitple -> multiple

r12a replied:
Shouldn't that be 'multiple languages' (with an -s) ?

aphillips replied:
🤦 See #79. Thanks for spotting these.
